src/rgw/rgw_user.cc

```diff
@@ -2462,8 +2465,11 @@ int RGWUserAdminOp_Subuser::remove(RGWRados *store, RGWUserAdminOpState& op_stat
 
 
   ret = user.subusers.remove(op_state, NULL);
-  if (ret < 0)
+  if (ret < 0) {
+    if (ret == -ENOENT)
+      ret = -ERR_NO_SUCH_SUBUSER;
```

Comment:
ERR_NO_SUCH_SUBUSER and the other similar error codes are returned to the rgw frontend and interpreted by the clients, so I think it's safe to use specific erros.